azurelinuxagent/common/protocol/wire.py

```diff
@@ -1490,13 +1490,15 @@ def parse(self, xml_text):
                 elif re.match(r'[-]+BEGIN.*CERTIFICATE[-]+', line):
                     begin_crt = True
                 elif re.match(r'[-]+END.*KEY[-]+', line):
+                    assert begin_prv
```

Comment:
We shouldn't use assertions yet since they would throw exceptions that the current code may not handle correctly (or at all). We can consider using optimized code to disable the asserts in production, but we can do that as a separate PR.

Reply:
fair enough, this can be changed to an exception